lib/plugins/aws/package/compile/events/apiGateway/index.js

```diff
@@ -20,15 +20,190 @@ const getMethodAuthorization = require('./lib/method/authorization');
 const getMethodIntegration = require('./lib/method/integration');
 const getMethodResponses = require('./lib/method/responses');
 
+function caseInsensitive(str) {
+  return { type: 'string', regexp: new RegExp(`^${str}$`, 'i').toString() };
+}
+
+const contentHandlingSchema = { enum: ['CONVERT_TO_BINARY', 'CONVERT_TO_TEXT'] };
+
+const allowedMethods = ['GET', 'POST', 'PUT', 'PATCH', 'OPTIONS', 'HEAD', 'DELETE', 'ANY'];
+const methodPattern = new RegExp(`^(?:\\*|${allowedMethods.join('|')})$`, 'i');
+const methodPathPattern = new RegExp(`^(?:\\*|(${allowedMethods.join('|')}) (\\/\\S*))$`, 'i');
+
+const requestParametersSchema = {
+  type: 'object',
+  additionalProperties: {
+    oneOf: [
+      { type: 'boolean' },
+      {
+        type: 'object',
+        properties: {
+          required: { type: 'boolean' },
+          mappedValue: { type: 'string' },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+};
+
+const authorizerSchema = {
+  oneOf: [
+    { type: 'string' },
+    caseInsensitive('aws_iam'),
+    { $ref: '#/definitions/awsArnString' },
+    {
+      type: 'object',
+      properties: {
+        arn: { $ref: '#/definitions/awsArn' },
+        authorizerId: { $ref: '#/definitions/awsCfInstruction' },
+        claims: { type: 'array', items: { type: 'string' } },
+        identitySource: { type: 'string' },
+        identityValidationExpression: { type: 'string' },
+        managedExternally: { type: 'boolean' },
+        name: { type: 'string' },
+        resultTtlInSeconds: { type: 'integer', minimum: 0, maximum: 3600 },
+        scopes: { type: 'array', items: { type: 'string' } },
+        type: {
+          oneOf: ['token', 'cognito_user_pools', 'request', 'aws_iam'].map(caseInsensitive),
+        },
+      },
+      required: [],
+      additionalProperties: false,
+    },
+  ],
+};
+
+const corsSchema = {
+  oneOf: [
+    { type: 'boolean' },
+    {
+      type: 'object',
+      properties: {
+        allowCredentials: { type: 'boolean' },
+        cacheControl: { type: 'string' },
+        headers: { type: 'array', items: { type: 'string' } },
+        maxAge: { type: 'integer', minimum: 1 },
+        methods: { type: 'array', items: { enum: allowedMethods } },
+        origin: { type: 'string' },
+        origins: {
+          type: 'array',
+          items: { type: 'string' },
+        },
+      },
+      oneOf: [{ required: ['origin'] }, { required: ['origins'] }],
```

Comment:
I think we need to also introduce `required: []`, otherwise we make passing origin required (either via `origin` or `origins` option)

Reply:
It seems to me from the code that indeed one of them is required. We also had this [manual validation](https://github.com/serverless/serverless/pull/8301/files#diff-872f8cac897abe18c7e8d00884f15926L28) that I removed which supports my theory.

Reply:
Indeed, it appears that  if someone uses an _object_ form, then not providing `origin` will fail.

Still that looks as apparent design bug, as by supporting `cors: true` we indicate that all properties have defaults. Intuitively object form should fall into `origin: '*'` if one was not set (it's e.g. the case in HTTP API, which has similar `cors` setting)

Anyway this is side issue (not a concern of this PR) so yes let's leave it as required.

Reply:
Oh I just saw [this](https://github.com/serverless/serverless/blob/master/lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js#L376). So it seems there is a `*` default for origin/origins, and that [validation error](url) probably couldn't be reached.